app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -57,6 +62,10 @@ def context
     user_session[:context] || 'authentication'
   end
 
+  def delivery_method
+    params[:delivery_method] || request.path.split('/').pop
```

Comment:
`.pop` -- seems weird to be doing a mutation, how about just `.last` ? (I understand it's an ephemeral array so it get re-created each time and doesn't actually mutate)